### PR TITLE
Added images and descriptions to docu of FreydCategory

### PR DESCRIPTION
**Pull request by HereAround**
I updated the documentation of weak kernel and weak kernel and added images for these descriptions. Does this close the issue number 185?

Also, I added descriptions and images for the following:
- Weak bi-fiber product
- Biased weak fiber product
- weak bi-pushout
- biased weak pushout

### CODE REVIEW

**Comment by kamalsaleh** [comment]
@HereAround 
Since this PR is still open, I would draw your attention that in the installation of the global function `INSTALL_FUNCTIONS_FOR_CAP_CATEGORY_OF_GRADED_ROWS` you install two other global functions `CAP_INTERNAL_WEAK_FIBER_PRODUCT_GRADED_ROWS` and `CAP_INTERNAL_WEAK_PUSHOUT_GRADED_ROWS`. This will cause error by any call of `CAPCategoryOfGradedRows` on any other ring, because global functions can not be reinstalled. I would recommend making them local-functions or moving them out.

**Comment by HereAround** [comment]
Thanks for pointing this out. @sebastianpos - thoughts?

**Comment by sebastianpos** [comment]
Yes, this is my mistake. The installation of the global methods was supposed to be outside of `INSTALL_FUNCTIONS_FOR_CAP_CATEGORY_OF_GRADED_ROWS`. I will fix this issue in a separate PR, since this PR was only supposed to be for documentation.

**Comment by HereAround** [comment]
Further adjustments of the images:
- size halved
- added \exists to all morphisms u( tau ) in the diagrams

File: FreydCategoriesForCAP/gap/FreydCategoriesForCAP.gd [hunk 1]
```diff
@@ -203,21 +219,37 @@ DeclareOperation( "AddWeakKernelLiftWithGivenWeakKernelObject",
...
+#! \begin{tikzpicture}
+#! \def\w{2};
+#! \node (A) at (0,0) {$A$};
+#! \node (B) at (\w,0) {$B$};
+#! \node (K) at (2*\w,0) {$K$};
+#! \node (T) at (2*\w,\w) {$T$};
+#! \draw[-latex] (A) to node[pos=0.45, above] {$\alpha$} (B);
+#! \draw[-latex] (B) to node[pos=0.45, above] {$\epsilon$} (K);
+#! \draw[-latex] (B) to node[pos=0.45, above left] {$\tau$} (T);
+#! \draw[-latex] (K) to node[pos=0.45, left] {$\exists u( \tau )$} (T);
```
Comment (sebastianpos) [inline FreydCategoriesForCAP/gap/FreydCategoriesForCAP.gd]:
```suggestion
#! \draw[-latex] (K) to node[pos=0.45, right] {$\exists u( \tau )$} (T);
```

File: FreydCategoriesForCAP/gap/FreydCategoriesForCAP.gd [hunk 2]
```diff
@@ -666,6 +761,38 @@ DeclareOperation( "AddUniversalMorphismIntoBiasedWeakFiberProductWithGivenBiased
 ##
 ####################################
 
+#! For a given pair of morphisms $(\alpha: A \rightarrow B, \beta \colon A \rightarrow C)$, a weak bi-pushout of $(\alpha, \beta)$ consists of three parts:
+#! * an object $P$,
+#! * morphisms $\iota_1: B \rightarrow P$, $\iota_2: C \rightarrow P$ such that $\iota_1 \circ \alpha \sim_{P,A} \iota_2 \circ \beta$,
```
Comment (sebastianpos) [inline FreydCategoriesForCAP/gap/FreydCategoriesForCAP.gd]:
```suggestion
#! * morphisms $\iota_1: B \rightarrow P$, $\iota_2: C \rightarrow P$ such that $\iota_1 \circ \alpha \sim_{A,P} \iota_2 \circ \beta$,
```

File: FreydCategoriesForCAP/gap/FreydCategoriesForCAP.gd [hunk 3]
```diff
@@ -666,6 +761,38 @@ DeclareOperation( "AddUniversalMorphismIntoBiasedWeakFiberProductWithGivenBiased
...
+#! * a dependent function $u$ mapping each pair $\tau = (\tau_1, \tau_2)$ of morphisms $\tau_1: B \rightarrow T$, $\tau_2: C \rightarrow T$ with the property $\tau_1 \circ \alpha \sim_{T,A} \tau_2 \circ \beta$ to a morphism $u(\tau): P \rightarrow T$ such that $u( \tau ) \circ \iota_1 \sim_{B,T} \tau_1$ and $u( \tau ) \circ \iota_2 \sim_{C,T} \tau_2$.
```
Comment (sebastianpos) [inline FreydCategoriesForCAP/gap/FreydCategoriesForCAP.gd]:
```suggestion
#! * a dependent function $u$ mapping each pair $\tau = (\tau_1, \tau_2)$ of morphisms $\tau_1: B \rightarrow T$, $\tau_2: C \rightarrow T$ with the property $\tau_1 \circ \alpha \sim_{A,T} \tau_2 \circ \beta$ to a morphism $u(\tau): P \rightarrow T$ such that $u( \tau ) \circ \iota_1 \sim_{B,T} \tau_1$ and $u( \tau ) \circ \iota_2 \sim_{C,T} \tau_2$.
```

File: FreydCategoriesForCAP/gap/FreydCategoriesForCAP.gd [hunk 4]
```diff
@@ -832,6 +959,37 @@ DeclareOperation( "AddDirectSumMorphismToWeakBiPushout",
 ##
 ####################################
 
+#! For a given pair of morphisms $(\alpha: A \rightarrow B, \beta: A \rightarrow C)$, a biased weak pushout  of $(\alpha, \beta)$ consists of three parts:
+#! * an object $P$,
+#! * a morphism $\iota: B \rightarrow P$ such that there exists a morphism $\delta: C \rightarrow P$ such that $\delta \circ \beta \sim_{P,A} \iota \circ \alpha$,
```
Comment (sebastianpos) [inline FreydCategoriesForCAP/gap/FreydCategoriesForCAP.gd]:
```suggestion
#! * a morphism $\iota: B \rightarrow P$ such that there exists a morphism $\delta: C \rightarrow P$ such that $\delta \circ \beta \sim_{A,P} \iota \circ \alpha$,
```

File: FreydCategoriesForCAP/gap/FreydCategoriesForCAP.gd [hunk 5]
```diff
@@ -832,6 +959,37 @@ DeclareOperation( "AddDirectSumMorphismToWeakBiPushout",
...
+#! * a dependent function $u$ mapping each $\tau: B \rightarrow T$, which admits a morphism $\mu \colon C \rightarrow T$ with $\mu \circ \beta \sim_{T,B} \tau \circ \alpha$, to a morphism $u(\tau):P \rightarrow T$ such that $u(\tau) \circ \iota \sim_{T,A} \tau$.
```
Comment (sebastianpos) [inline FreydCategoriesForCAP/gap/FreydCategoriesForCAP.gd]:
```suggestion
#! * a dependent function $u$ mapping each $\tau: B \rightarrow T$, which admits a morphism $\mu \colon C \rightarrow T$ with $\mu \circ \beta \sim_{B,T} \tau \circ \alpha$, to a morphism $u(\tau):P \rightarrow T$ such that $u(\tau) \circ \iota \sim_{A,T} \tau$.
```

**Comment by HereAround** [comment]
All corrections look good. Thanks for the suggestions - all accepted.

Regarding https://github.com/homalg-project/CAP_project/issues/185, I have updated the definitions of weak (co)kernel in the following two commits:

- https://github.com/homalg-project/CAP_project/pull/297/commits/28a2da9fd5cb834baf8b6836adbcf323f3857259
- https://github.com/homalg-project/CAP_project/pull/297/commits/44c98717a5481013093c57dd217dce6c25bb23c0

@sebastianpos , please check if the wording is correct. Are there other methods which need to be adjusted?

**Comment by sebastianpos** [comment]
I didn't mean the description of weak kernels in the introduction, but the actual documentation of the GAP operations `WeakKernelObject`, `WeakKernelEmbedding`, etc.

By the way, could you edit the head comment that you made in this PR?
I think that the question you posed "Does this close [...]" actually will trigger github to close the issue once I merge this PR.

**Comment by HereAround** [comment]
Ok. I also adjusted the head commit.

**Pull request merged**